granary/test/test_flickr.py

```diff
@@ -0,0 +1,651 @@
+    }],
+  },
+}
+
+CONTACTS_PHOTOS_ACTIVITIES = [{
+  'verb': 'post',
+  'actor': {'numeric_id': '5555'},
+  'created': '2013-06-08 03:20:48',
+  'url': 'https://www.flickr.com/photos/5555/1234',
+  'object': {
+    'content': 'First Photo\n',
+    'displayName': 'First Photo',
+    'created': '2013-06-08 03:20:48',
+    'url': 'https://www.flickr.com/photos/5555/1234',
+    'image': {
+      'url': 'https://farm4.staticflickr.com/99/1234_22_b.jpg'
+    },
+    'published': '2013-06-09T17:40:34',
+    'id': 'tag:flickr.com:1234',
+    'tags': [{
+      'url': 'https://www.flickr.com/search?tags=tag1',
+      'displayName': 'tag1',
+      'objectType': 'hashtag'
+    }, {
+      'url': 'https://www.flickr.com/search?tags=tag2',
+      'displayName': 'tag2',
+      'objectType': 'hashtag'
+    }],
+    'objectType': 'photo'
+  },
+  'id': 'tag:flickr.com:1234',
+  'flickr_id': '1234',
+  'published': '2013-06-09T17:40:34'
+}, {
+  'verb': 'post',
+  'actor': {'numeric_id': '6666'}, 'created': '2010-11-27 12:54:33',
+  'url': 'https://www.flickr.com/photos/6666/2345',
+  'object': {
+    'content': 'Second Photo\n',
+    'displayName': 'Second Photo',
+    'created': '2010-11-27 12:54:33',
+    'url': 'https://www.flickr.com/photos/6666/2345',
+    'image': {
+      'url': 'https://farm4.staticflickr.com/88/2345_33_b.jpg'
+    },
+    'published': '2010-12-03T01:16:19',
+    'id': 'tag:flickr.com:2345',
+    'tags': [{
+      'url': 'https://www.flickr.com/search?tags=tag1',
+      'displayName': 'tag1',
+      'objectType': 'hashtag'
+    }, {
+      'url': 'https://www.flickr.com/search?tags=tag2',
+      'displayName': 'tag2',
+      'objectType': 'hashtag'
+    }],
+    'objectType': 'photo'
+  },
+  'id': 'tag:flickr.com:2345',
+  'flickr_id': '2345',
+  'published': '2010-12-03T01:16:19'
+}]
+
+ACTOR = {
+  'objectType': 'person',
+  'displayName': 'Kyle Mahan',
+  'image': {
+    'url': 'https://farm5.staticflickr.com/4068/buddyicons/39216764@N00.jpg',
+  },
+  'id': 'tag:flickr.com:kindofblue115',
+  'location': {'displayName': 'San Diego, CA, USA'},
+  'description': 'Trying a little bit of everything',
+  'url': 'https://kylewm.com/',
+  'urls': [
+    {'value': 'https://www.flickr.com/people/kindofblue115/'},
+    {'value': 'https://www.flickr.com/people/kindofblue115/contacts/'},
+    {'value': 'https://kylewm.com/'},
+  ],
+  'username': 'kindofblue115',
+  'numeric_id': '39216764@N00',
+}
+
+PROFILE_HTML = """
+<html>
+  <body>
+    <a href='https://www.flickr.com/people/kindofblue115/' rel='me'>Profile</a>
+    <a href='https://www.flickr.com/people/kindofblue115/contacts/' rel='me'>Contacts</a>
+    <a href='https://kylewm.com/' rel='me'>kylewm.com</a>
+  </body>
+</html>
+"""
+
+
+class FlickrTest(testutil.TestCase):
+
+  def setUp(self):
+    super(FlickrTest, self).setUp()
+    appengine_config.FLICKR_APP_KEY = 'fake'
+    appengine_config.FLICKR_APP_SECRET = 'fake'
+    od_appengine_config.FLICKR_APP_KEY = 'fake'
```

Comment:
minor: interesting. the other tests don't need to do this, including twitter. does flickr?